backend/controller/controller.go

```diff
@@ -1393,6 +1393,9 @@ func (s *Service) executeAsyncCalls(ctx context.Context) (time.Duration, error)
 	resp, err := s.callWithRequest(ctx, connect.NewRequest(req), optional.None[model.RequestKey](), s.config.Advertise.String())
 	var callResult either.Either[[]byte, string]
 	failed := false
+	// Queue depth is queried at acquisition time, which means it includes the async
+	// call that was just executed, so decrement the value accordingly.
+	queueDepth := call.QueueDepth - 1
```

Comment:
`callWithRequest` is also used for ingress :/

Reply:
Ah yeah, just meant having this line above the `callWithRequest` line above to avoid the decrement. Definitely a nit

Reply:
Ooooh gotcha! Sorry totally misunderstood! We would need the decrement anyway because QueueDepth is calculated all the way up at `AcquireAsyncCall()`. I'll move it up so it's easier to follow.